addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -155,25 +162,62 @@ private void updateThingFromDevice(Thing thing, DeviceModel device) {
         }
         if (device.getPresent() == 1) {
             thing.setStatusInfo(new ThingStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE, null));
-            logger.debug("about to update {} from {}", thing.getUID(), device);
-            if (device.isTempSensor()) {
-                Channel channel = thing.getChannel(CHANNEL_TEMP);
-                this.updateState(channel.getUID(), new DecimalType(device.getTemperature().getCelsius()));
+            logger.debug("about to update thing {} from device {}", thing.getUID(), device);
+            if (device.isTempSensor() && device.getTemperature() != null) {
```

Comment:
Is this (the null check) a scenario which can happen, sounds like a programming error to me?